rust/lancedb/src/query.rs

```diff
@@ -862,6 +903,66 @@ impl VectorQuery {
         self.use_index = false;
         self
     }
+
+    pub async fn execute_hybrid(&self) -> Result<SendableRecordBatchStream> {
+        // clone query and specify we want to include row IDs, which can be needed for reranking
+        let fts_query = self.base.clone().with_row_id();
+        let mut vector_query = self.clone().with_row_id();
+
+        vector_query.base.full_text_search = None;
+        let (fts_results, vec_results) = try_join!(fts_query.execute(), vector_query.execute())?;
+
+        let fts_results = fts_results.try_collect::<Vec<_>>().await?;
+        let vec_results = vec_results.try_collect::<Vec<_>>().await?;
```

Comment:
To make them run concurrently, we'll want to poll the streams concurrently too.
```suggestion
        let (fts_results, vec_results) = try_join!(
            fts_results.try_collect::<Vec<_>>(),
            vec_results.try_collect::<Vec<_>>(),
        )?;
```

Reply:
done good call